### PR TITLE
[Android] Handling Permission Appropriately in Jetpack Compose

### LINKED ISSUE #5571: [Android] Handling Permission Appropriately in Jetpack Compose
#### Introduction
App permissions help support user privacy by protecting access to restricted data such as system state and Restricted actions, such as recording audio. 
With Jetpack Compose there are some changes in the approach and how we can request permissions.
Most developers do not handle these permissions correctly. In this article, I will explain the easiest ways in which one can handle single or multiple permissions in the android app at runtime.  

#### Key takeaways 
- What is permission handling
- What are runtime permissions
- How to implement single and multiple permissions at runtime

#### Article quality
This article will focus on showing a learner how to handle permissions correctly and with ease. It will be beginner-friendly to those who are venturing into Jetpack Compose and easy to follow along

#### References
- Permissions Handling in Android

### PR DESCRIPTION
**Pull request by tanscy-cassie**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5571

### CODE REVIEW

File: content/articles/permission-handling-in-jetpack-compose/index.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
+In android, permission handling can be done by the help of `permissions API`. With the help of Google's accompanist libraries, things are made more easy. 
+In this tutorial, we will learn on how we can handle permissions in Jetpack compose.
+
```
Comment (BrianaNzivu) [inline content/articles/permission-handling-in-jetpack-compose/index.md:3]: You can add the table of contents to this article here.

File: content/articles/permission-handling-in-jetpack-compose/index.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+In android, permissions define what an app can access in a user's phone. Due to security measures, an app cannot access some of the phone's data. This requires the app to request the user to allow or deny the app access. Accepting the permissions will allow the app to access some of the data like contacts, SMS etc.
+
+They are used whenever the app needs user authorization to access hardware or data that is not accessible by default.
+### Step 1: Creating a new compose project
+To create a new Compose project, launch Android studio and select `New Project` then `Compose Activity`. Name it `PermissionsDemo` and click on finish to build the project. Wait till the building is finished.
+
+![Create Project](/engineering-education/permission-handling-in-jetpack-compse/create-project.png)
+
+### Step 2: Setup the dependency
+Add this dependency in the `build.gradle` file.
```
Comment (BrianaNzivu) [inline content/articles/permission-handling-in-jetpack-compose/index.md:33]: List which dependencies you are adding i.e add the permissions dependency

File: content/articles/permission-handling-in-jetpack-compose/index.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+        setContent {
+            PermissionHandlingComposeDEmoTheme {
+                //SinglePermission()
+                MultiplePermissions()
+            }
+        }
+    }
+}
+```
+
```
Comment (BrianaNzivu) [inline content/articles/permission-handling-in-jetpack-compose/index.md:341]: You can link the GitHub repo with the sample code of this project here. Also, you can have a downloadable sample APK which you can host on Google drive.

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @tanscy-cassie 👋 .
 
My name is Briana, I will be reviewing your article. I promise to make this review as fast and as swift as possible, with your help of course 😄 .

Good job on the article. It is a very interesting read. Keep up the good work 👍 .

Just a few changes I need you to make as we continue with the preliminary revisions

In case of any question or help needed please do not hesitate to ping me here or on Slack.
Also, notify me when you are done.

Regards.

**Comment by tanscy-cassie** [comment]
@BrianaNzivu i have made the requested changes

File: content/articles/permission-handling-in-jetpack-compose/index.md [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+### Goals
+By the end of this tutorial, the reader will be able to:
+- Have an understanding of what permissions are and when they are used
+- Use the most effective way to handle permissions in Jetpack compose
+- Implement single and multiple permissions in Jetpack compose
+
+### Terminologies
+- `DisposableEffect` - A side-effect handler for side-effects which when keys change or when a composable leaves composition.
+- `Lifecycle` - This is an abstract class associated with the android lifespan which allows an object to detect states and respond accordingly.
+- `Rationale` -
```
Comment (BrianaNzivu) [inline content/articles/permission-handling-in-jetpack-compose/index.md:30]: This lacks a definition.

File: content/articles/permission-handling-in-jetpack-compose/index.md [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+
+They are used whenever the app needs user authorization to access hardware or data that is not accessible by default.
+### Step One: Creating a new compose project
+To create a new Compose project, launch Android studio and select `New Project` then `Compose Activity`. Name it `PermissionsDemo` and click on finish to build the project. Wait till the building is finished.
+
+![Create Project](/engineering-education/permission-handling-in-jetpack-compse/create-project.png)
+
+### Step Two: Setup the dependency
+Here we are going to add the `accompanist-permissions` dependency.
+Add this dependency in the `build.gradle` file.
```
Comment (BrianaNzivu) [inline content/articles/permission-handling-in-jetpack-compose/index.md:45]: You can specify if this build.gradle file is either app or project level.

File: content/articles/permission-handling-in-jetpack-compose/index.md [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+```gradle
+implementation 'com.google.accompanist:accompanist-permissions:0.21.1-beta'
+```
+In this tutorial, we will use the [Google Accompanist]() library to handle permissions. There are some other ways you can use to handle but they are much more complicated. Using the `accompanist` library will make it simpler.
```
Comment (BrianaNzivu) [inline content/articles/permission-handling-in-jetpack-compose/index.md:49]: Link Google Accompanist page.

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @tanscy-cassie 👋 .
Just minor corrections as I proceed with the tech review.

Regards.

**Comment by tanscy-cassie** [comment]
@BrianaNzivu I am done with the requested changes ✔️

**Review by BrianaNzivu — APPROVED** [review]
Hey @tanscy-cassie   👋 .

Very good job on the article. It is a very interesting read 👏 .

Just a few things I'd like to point out:
-  It is always advisable to submit an article that is well polished. By this, I mean having an article that does not have too many grammatical corrections and functional code. I suggest that you check out our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) page which has links to various resourceful tools for writing. 
- Lastly please go through ALL the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

**Pull request merged**